CMakeLists.txt

```diff
@@ -70,6 +107,10 @@ file(GLOB ${PROJECT_NAME}_SOURCES src/*.cpp)
 nanobind_add_module(nanoeigenpy NB_STATIC NB_SUPPRESS_WARNINGS ${nanoeigenpy_SOURCES} ${nanoeigenpy_HEADERS})
 target_link_libraries(nanoeigenpy PUBLIC nanoeigenpy_headers)
 
+if(BUILD_WITH_CHOLMOD_SUPPORT)
+  target_link_libraries(nanoeigenpy PUBLIC CHOLMOD::CHOLMOD)
```

Comment:
This is the wrong target. It should be
```cmake
  target_link_libraries(nanoeigenpy_headers PUBLIC CHOLMOD::CHOLMOD)
```
Since the headers target is the one that gets distributed and that CMake users consume.